sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/QueueClientBuilder.java

```diff
@@ -0,0 +1,338 @@
+import java.util.ServiceLoader;
+
+/***
+ * The builder to create {@link QueueReceiverAsyncClient} and {@link QueueSenderAsyncClient}.
+ */
+@ServiceClientBuilder(serviceClients = { QueueReceiverAsyncClient.class, QueueSenderAsyncClient.class})
+public final class QueueClientBuilder {
+
+    private static final String AZURE_SERVICE_BUS_CONNECTION_STRING = "AZURE_SERVICE_BUS_CONNECTION_STRING";
+    private static final AmqpRetryOptions DEFAULT_RETRY =
+        new AmqpRetryOptions().setTryTimeout(ServiceBusClientConstants.OPERATION_TIMEOUT);
+
+    private static final String SERVICEBUS_PROPERTIES_FILE = "azure-messaging-servicebus.properties";
+    private static final String NAME_KEY = "name";
+    private static final String VERSION_KEY = "version";
+    private static final String UNKNOWN = "UNKNOWN";
+
+    private final ClientLogger logger = new ClientLogger(QueueClientBuilder.class);
+
+    private ProxyOptions proxyOptions;
+    private TokenCredential credentials;
+    private Configuration configuration;
+    private AmqpRetryOptions retryOptions;
+    private Scheduler scheduler;
+    private AmqpTransportType transport = AmqpTransportType.AMQP;
+    private String fullyQualifiedNamespace;
+    private String queueName;
+    private ServiceBusConnectionProcessor servicerBusConnectionProcessor;
+    private final String connectionId;
+
+    /**
+     * Creates a new instance with the default transport {@link AmqpTransportType#AMQP}.
+     */
+    public QueueClientBuilder() {
+        this.connectionId = StringUtil.getRandomString("MF");
+    }
+
+    /**
+     *
+     * @param connectionString to connect to Queue.
+     * @return The updated {@link QueueClientBuilder} object.
+     */
+    public QueueClientBuilder connectionString(String connectionString) {
+        final ConnectionStringProperties properties = new ConnectionStringProperties(connectionString);
+        final TokenCredential tokenCredential;
+        try {
+            tokenCredential = new ServiceBusSharedKeyCredential(properties.getSharedAccessKeyName(),
+                properties.getSharedAccessKey(), ServiceBusClientConstants.TOKEN_VALIDITY);
+        } catch (Exception e) {
+            throw logger.logExceptionAsError(
+                new AzureException("Could not create the ServiceBusSharedKeyCredential.", e));
+        }
+        this.fullyQualifiedNamespace = properties.getEndpoint().getHost();
+        this.queueName = properties.getEntityPath();
+        return credential(properties.getEndpoint().getHost(), properties.getEntityPath(), tokenCredential);
+    }
+
+    /**
+     *
+     * @param fullyQualifiedNamespace for the Service Bus.
+     * @param queueName The name of the queue.
+     * @param credential {@link TokenCredential} to be used for authentication.
+     * @return The updated {@link QueueClientBuilder} object.
+     */
+    public QueueClientBuilder credential(String fullyQualifiedNamespace, String queueName, TokenCredential credential) {
```

Comment:
Probably a good point for us to sync on this - I think we're calling the `fullyQualifiedNamespace` just `host`.

Reply:
I got impression that EH had issues where useres were confused with `host` . I am open to idea but we should get review by EH team.